httpclient/httpclient.go

```diff
@@ -26,3 +29,30 @@ func CloneHTTPHeader(in http.Header) http.Header {
 	}
 	return out
 }
+
+// MustHTTPClient wraps delegate transports with logging, metrics, rate limiting, retryable, user agent, request id
+// and panics if any error occurs.
+func MustHTTPClient(cfg *Config, userAgent, reqType string, delegate http.RoundTripper) *http.Client {
```

Comment:
Let’s provide both methods:
- NewHTTPClient: This method returns an error, allowing callers to handle it gracefully.
- MustHTTPClient: This method panics if an error occurs, providing a convenient option for cases where error handling is not necessary.

This approach ensures flexibility for different use cases.